language/compiler/ir-to-bytecode/syntax/src/lexer.rs

```diff
@@ -148,6 +150,18 @@ impl<'input> Lexer<'input> {
         self.token = (start_offset, Token(tok, result), end_offset);
         Ok(())
     }
+
+    pub fn replace_tok(
+        &mut self,
+        tok: Tok,
+        delta: usize,
+    ) -> Result<(), ParseError<usize, anyhow::Error>> {
+        self.previous_end += delta;
```

Comment:
I liked your idea from yesterday better (or maybe I just had something different in mind). This seems like it is a combination of the 2 approaches that we had discussed. Unless you discovered an unexpected complication, I think you can avoid changing "previous_end", "consumed", and "self.token.0". Instead, just add the delta to "self.token.2" and set the new token to "Token(tok, self.text)".

Reply:
It could be that I misunderstood what each field means. Let me study your existing code more and double check on this.